lib/hawkular/operations/operations_api.rb

```diff
@@ -33,9 +33,31 @@ class Client < Hawkular::BaseClient
     # helper for parsing the "OperationName=json_payload" messages
     class WebSocket::Frame::Data
       def to_msg_hash
-        chunks = split('=', 2)
+        operation_name, json = split('=', 2)
 
-        { operationName: chunks[0], data: JSON.parse(chunks[1]) }
+        # Check if there is a zip file following JSON.
+        # This check is done only in the first 100KB, hoping it's unlikely to
+        # have such large amount of JSON before a zip file is attached.
+        magic_bits = ["\x50\x4B\x03\x04", "\x50\x4B\x05\x06", "\x50\x4B\x07\x08"]
+        search_chunk = json[0, 102_400]
```

Comment:
According to [this code](https://github.com/josejulio/hawkular-agent/blob/f3b9238af581cea8b5c2f3fe5b7dab41ed58a581/hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/cmd/FeedCommProcessor.java#L289), when binary data is present, the `content-type` gets changed to `application/vnd.okhttp.websocket+binary`.

Maybe we could search only when binary data is know to be present?